woocommerce/woocommerce-ios
# [Mobile Payments] Fix Tap to Pay auto reconnection on fresh launch

Closes: #11052 


## Description


Previously, the auto-reconnection to the Tap to Pay system was not working on first launch; no connection was attempted. This reconnection is intended to save merchants time when taking a payment with TTP, by avoiding the wait for connection.

It was failing because in `reconnectIfNeeded` it checks for `case .completed = onboardingCache.value` in the guard. On first launch, this was always `nil`, because it hadn’t finished refreshing. Both the refresh and the reconnection are triggered by app launch and app foreground.

In this PR, we wait until the onboarding cache has finished refreshing, before we auto-reconnect to the Tap to Pay system.

## Testing instructions


Using a US or UK based WooPayments store:

1. Launch the app from Xcode
2. Take a Tap to Pay transaction, to ensure you're eligible for TTP reconnection on this device.
3. Force quit the app
4. Launch the app again (from Xcode)
5. Wait 5-15s to allow reconnection to complete
6. Observe that the console output includes `card_reader_connection_success` tracks event (or `card_reader_connection_failed`)
7. Navigate to `Menu > Payments > Try Out Tap to Pay on iPhone` 
8. Observe that you're on the `Setup complete` screen shows, indicating that the app automatically connected.

N.B. if the reconnection fails, you'll see the tracks event for failure, but the `Set up Tap to Pay on iPhone` screen when you navigate to `Try out TTP`

## Screenshots


### Before

https://github.com/woocommerce/woocommerce-ios/assets/2472348/ad277295-1ecd-484b-b04f-fbfad301beaf


### After

https://github.com/woocommerce/woocommerce-ios/assets/2472348/d810ee0c-4094-440d-a297-54d5d855b284



---
